feat(tailscale)!: Update tailscale to Oauth (#7741)
cloudquery/cloudquery: website/pages/docs/plugins/sources/tailscale/overview.mdx

## Authentication

In order to fetch information from Tailscale, `cloudquery` needs to be authenticated.

> The suggested way is to use an [OAuth client](https://tailscale.com/kb/1215/oauth-clients/), the previous way of
> using An [API key](https://login.tailscale.com/admin/settings/keys) is also supported.

Reviewer: ```suggestion
The suggested way is to use an [OAuth client](https://tailscale.com/kb/1215/oauth-clients/). The previous way of
using an [API key](https://login.tailscale.com/admin/settings/keys) is also supported.
```

## Spec

This is the (nested) spec used by the Tailscale source plugin.

- `client_id` (string, required):
  Client ID of the [OAuth Client](https://tailscale.com/kb/1215/oauth-clients/).

- `client_secret` (string, required):
  Client secret of the [OAuth Client](https://tailscale.com/kb/1215/oauth-clients/) (Make sure to use environment variable
  expansion and not hard code the keys in the config YAML).

- `tailnet`  (string, required):
  This is your Tailscale tailnet name (also known as organization name).

- `api_key` (string, optional):
  This way of authentication is not recommended. An API key to access Tailscale resources.
  This can be obtained from [Tailscale Keys Settings Page](https://login.tailscale.com/admin/settings/keys).

- `endpoint_url` (string, optional. Default: not used):
  Endpoint URL to make the API requests to.

## Example

> This example syncs from Tailscale to a Postgres destination, using `api_key` authentication.

Reviewer: ```suggestion
This example syncs from Tailscale to a Postgres destination, using OAuth authentication.
```

The (top level) source spec section is described in the [Source Spec Reference](/docs/reference/source-spec).

```yaml copy
kind: source
# Common source-plugin configuration
spec:
  name:    tailscale
  path:    cloudquery/tailscale
  version: "VERSION_SOURCE_TAILSCALE"
  tables: [ "*" ]
  destinations: [ "postgresql" ]

  # plugin specific configuration
  spec:

>     client_id:    "<YOUR_API_KEY_HERE>"

Reviewer: ```suggestion
    client_id:    "<YOUR_CLIENT_ID_HERE>"
```
